applications/IgaApplication/custom_conditions/support_penalty_condition.h

```diff
@@ -0,0 +1,264 @@
+        * @param rLeftHandSideMatrix the condition left hand side matrix
+        * @param rCurrentProcessInfo the current process info
+        */
+        void CalculateLeftHandSide(
+            MatrixType& rLeftHandSideMatrix,
+            const ProcessInfo& rCurrentProcessInfo) override
+        {
+            const SizeType mat_size = GetGeometry().size() * 3;
+
+            VectorType right_hand_side_vector;
+
+            if (rLeftHandSideMatrix.size1() != mat_size && rLeftHandSideMatrix.size2())
+                rLeftHandSideMatrix.resize(mat_size, mat_size);
+            noalias(rLeftHandSideMatrix) = ZeroMatrix(mat_size, mat_size);
+
+            CalculateAll(rLeftHandSideMatrix, right_hand_side_vector,
+                rCurrentProcessInfo, true, false);
+        }
+
+        /**
+         * @brief This function provides a more general interface to the element.
+         * @details It is designed so that rLHSvariables and rRHSvariables are
+         *          passed to the element thus telling what is the desired output
+         * @param rLeftHandSideMatrix container with the output Left Hand Side matrix
+         * @param rRightHandSideVector container for the desired RHS output
+         * @param rCurrentProcessInfo the current process info instance
+         */
+        void CalculateLocalSystem(
+            MatrixType& rLeftHandSideMatrix,
+            VectorType& rRightHandSideVector,
+            const ProcessInfo& rCurrentProcessInfo) override
+        {
+            const SizeType mat_size = GetGeometry().size() * 3;
+
+            if (rRightHandSideVector.size() != mat_size)
+                rRightHandSideVector.resize(mat_size);
+            noalias(rRightHandSideVector) = ZeroVector(mat_size);
+
+            if (rLeftHandSideMatrix.size1() != mat_size)
+                rLeftHandSideMatrix.resize(mat_size, mat_size);
+            noalias(rLeftHandSideMatrix) = ZeroMatrix(mat_size, mat_size);
+
+            CalculateAll(rLeftHandSideMatrix, rRightHandSideVector,
+                rCurrentProcessInfo, true, true);
+        }
+
+        /**
+        * @brief Sets on rResult the ID's of the element degrees of freedom
+        * @param rResult The vector containing the equation id
+        * @param rCurrentProcessInfo The current process info instance
+        */
+        void EquationIdVector(
+            EquationIdVectorType& rResult,
+            const ProcessInfo& rCurrentProcessInfo
+        ) const override;
+
+        /**
+        * @brief Sets on rConditionDofList the degrees of freedom of the considered element geometry
+        * @param rElementalDofList The vector containing the dof of the element
+        * @param rCurrentProcessInfo The current process info instance
+        */
+        void GetDofList(
+            DofsVectorType& rElementalDofList,
+            const ProcessInfo& rCurrentProcessInfo
+        ) const override;
+
+        /**
+        * This functions calculates both the RHS and the LHS

```

Comment:
@brief is missing and no double dots are needed

Reply:
I killed it completely 😎